pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentLoader.java

```diff
@@ -95,35 +96,52 @@ public static ImmutableSegment load(File indexDir, IndexLoadingConfig indexLoadi
    * modify the segment like to convert segment format, add or remove indices.
    */
   public static ImmutableSegment load(File indexDir, IndexLoadingConfig indexLoadingConfig, @Nullable Schema schema,
```

Comment:
where is this load method being used now? All calls I saw in BaseDataTableManager were for the new load method

Reply:
In few test cases and RawIndexConverter. 

There are other load() methods above for test or non-test classes. Those load() variants can be refactored to use the new load(SegmentDirectory), but there are still some boilerplate codes which I kept in this load() method.